validator/sawtooth_validator/journal/block_sender.py

```diff
@@ -37,6 +40,24 @@ def __init__(self, completer, gossip):
         self._completer = completer
         self._gossip = gossip
 
-    def send(self, block):
-        self._gossip.broadcast_block(block)
+    def send(self, block, injected_batch_ids=None):
+        self._gossip.broadcast_block(
+            self._remove_batches(block, keep_batches=injected_batch_ids))
         self._completer.add_block(block)
+
+    def _remove_batches(self, block, keep_batches):
+        """Returns a copy of the block without the non-injected
+        batches, which other validators are likely to already
+        have. This reduces the size of the broadcasted block.
+        """
+        block_header = BlockHeader()
+        block_header.ParseFromString(block.header)
+        block_builder = BlockBuilder(block_header)
+        block_builder.set_signature(block.header_signature)
+
+        if keep_batches is not None:
```

Comment:
I would move this check up to the send function and only call `_remove_batches` when there are batches to remove.